baselib.py

```diff
@@ -505,8 +510,8 @@ def set_motion_paths_netcdf(context):
                         rotvec_norm[0], rotvec_norm[1], rotvec_norm[2] ))
                 obj.keyframe_insert(data_path = "rotation_axis_angle")
         elif obj.mbdyn.parametrization == 'MATRIX':
-            for frame in range(scene.frame_end + 1):
-                scene.frame_current = frame
+            for frame in range(scene.frame_start, scene.frame_end):
+                scene.frame_current += 1
```

Comment:
This solution will not work for NetCDF output.
When using text file output, at each timestep all the nodes' objects in the Blender scene are moved at the same time. 

When using NetCDF output though, the entire time history of a single node is imported, then we move on to the 
next node and so on till we set all the motion paths for all the nodes. 

You should reset the `scene.frame_current` variable every time you change the current node.